WORKSPACE

```diff
@@ -119,12 +129,14 @@ stack_snapshot(
         "ghc-boot-th",
         "pretty",
         "transformers",
-    ],
+    ] + (["linear-base"] if ghc_version == "9.0.1" else []),
     vendored_packages =
       { "singletons": "@singletons//:singletons"
       , "th-desugar": "@th-desugar//:th-desugar"
-      },
-    snapshot = "lts-16.5",
+      } if ghc_version == "8.10.1" else {},
+    snapshot = "lts-16.5" if ghc_version == "8.10.1" else None,
+    local_snapshot = "//:snapshot-9.0.1.yaml" if ghc_version == "9.0.1" else None,
+    stack = "@stack//:bin/stack" if ghc_version == "9.0.1" else None,
```

Comment:
Why not use that `stack` for all GHC versions?

Reply:
This is changing the way stack behaves from what `rules_haskell` expects. I'm keeping it only when necessary for that reason. 

Also, it is striking me that since `stack ls dependencies` is failing in CI only for the 9.0.1 case, this bit of configuration might not be robust enough.

Reply:
Makes sense, thanks for clarifying!

Reply:
I made the name of the repo more explicit to indicate that it is ignoring global hints.